### PR TITLE
Updated Person model attributes that needs to be unique

### PR DESCRIPTION
**Pull request by cwq2326**
Updated behaviors for attributes (phone, matriculation number, email; removed name attribute) in the Person model so that all three attributes must be different in order for it to be unique. Previously, only the name attribute affects this behavior. In addition, the test cases are also updated to test different combinations of these attributes. In addition, the UG is also updated to reflect the new changes.

The reason for this change is to reflect more accurately on what attributes is defined as unique for a student. For example, students can have the same name but they cannot have the same email, matriculation number or phone number. This change is performed by modifying the behavior of the isSamePerson method to check if the person being added has any of the same three attributes. 

The purpose of this PR is for feedback for an initial implementation.

This is part of a greater set of changes and is dependent on #62 as it is modifying codes in the same class.

Ran JUnit to test the new behavior of the Person model, and also added new test cases such as having different combinations of similiar and different unique attributes, including different casing.

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 1]
```diff
@@ -31,23 +32,78 @@ public void isSamePerson() {
...
+        // matriculation number differs in case, all other attributes same -> returns true
+        editedBob = new PersonBuilder(BOB).withMatricNumber(VALID_MATRICULATION_NUMBER_BOB.toUpperCase()).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        editedBob = new PersonBuilder(BOB).withMatricNumber(VALID_MATRICULATION_NUMBER_BOB.toLowerCase()).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        // First character is lower case last character is upper case eg. a0253625M
+        String editedMatriculationNumber = VALID_MATRICULATION_NUMBER_BOB.substring(0, 1).toUpperCase()
+                + VALID_MATRICULATION_NUMBER_BOB.substring(1).toLowerCase();
```
Comment (DannyDakota) [inline src/test/java/seedu/address/model/person/PersonTest.java:82]: The comment for this should be the first character to uppercase and the rest to lowercase 👍

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 2]
```diff
@@ -31,23 +32,78 @@ public void isSamePerson() {
...
+
+        editedBob = new PersonBuilder(BOB).withMatricNumber(editedMatriculationNumber).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        // First character is upper case last character is lower case eg. A0253625m
```
Comment (DannyDakota) [inline src/test/java/seedu/address/model/person/PersonTest.java:87]: opposite for this 👍

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 3]
```diff
@@ -31,23 +32,78 @@ public void isSamePerson() {
...
+        editedMatriculationNumber = VALID_MATRICULATION_NUMBER_BOB.substring(0, 1).toLowerCase()
+                + VALID_MATRICULATION_NUMBER_BOB.substring(1).toUpperCase();
+        editedBob = new PersonBuilder(BOB).withMatricNumber(editedMatriculationNumber).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        // name has trailing spaces, all other attributes same -> returns false
```
Comment (DannyDakota) [inline src/test/java/seedu/address/model/person/PersonTest.java:93]: Incorrect comment for this? "_email_ has trailing spaces, all other attribute same -> returns _true_" 😄

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 4]
```diff
@@ -31,23 +32,78 @@ public void isSamePerson() {
...
+        String emailWithTrailingSpaces = VALID_EMAIL_BOB + " ";
+        editedBob = new PersonBuilder(BOB).withEmail(emailWithTrailingSpaces).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        // name has trailing spaces, all other attributes same -> returns false
+        String phoneWithTrailingSpaces = VALID_PHONE_BOB + " ";
+        editedBob = new PersonBuilder(BOB).withName(phoneWithTrailingSpaces).build();
```
Comment (DannyDakota) [inline src/test/java/seedu/address/model/person/PersonTest.java:100]: Check this test case again? the variable phoneWithTrailingSpaces is being tested but the method withName is being used? Also, comment needs to be updated 👍

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 5]
```diff
@@ -31,23 +32,78 @@ public void isSamePerson() {
...
+        assertTrue(BOB.isSamePerson(editedBob));
 
         // name has trailing spaces, all other attributes same -> returns false
-        String nameWithTrailingSpaces = VALID_NAME_BOB + " ";
-        editedBob = new PersonBuilder(BOB).withName(nameWithTrailingSpaces).build();
-        assertFalse(BOB.isSamePerson(editedBob));
+        String matriculationNumberWithTrailingSpaces = VALID_MATRICULATION_NUMBER_BOB + " ";
+        editedBob = new PersonBuilder(BOB).withName(matriculationNumberWithTrailingSpaces).build();
```
Comment (DannyDakota) [inline src/test/java/seedu/address/model/person/PersonTest.java:105]: same as above! 😄

**Review by DannyDakota — CHANGES_REQUESTED** [review]
LGTM! only need to revisit some test cases and change some comments!

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 1 above)
Comment (cwq2326) [reply src/test/java/seedu/address/model/person/PersonTest.java:82]: Thanks for pointing out 😄

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 2 above)
Comment (cwq2326) [reply src/test/java/seedu/address/model/person/PersonTest.java:87]: Thanks for pointing out 😄

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 3 above)
Comment (cwq2326) [reply src/test/java/seedu/address/model/person/PersonTest.java:93]: Thanks for pointing out 😄

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 4 above)
Comment (cwq2326) [reply src/test/java/seedu/address/model/person/PersonTest.java:100]: Thanks for pointing out 😄

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 5 above)
Comment (cwq2326) [reply src/test/java/seedu/address/model/person/PersonTest.java:105]: Thanks for pointing out 😄

**Review by DannyDakota — APPROVED** [review]
LGTM!

**Pull request merged**